kernel/src/hil/time.rs

```diff
@@ -175,3 +198,254 @@ pub trait TimerClient {
     /// Callback signaled when the timer's clock reaches the specified interval.
     fn fired(&self);
 }
+
+pub trait Ticks: Clone + Copy + From<u32> {
+    fn into_usize(self) -> usize;
+    fn into_u32(self) -> u32;
+
+    fn wrapping_add(self, other: Self) -> Self;
+    fn wrapping_sub(self, other: Self) -> Self;
+
+    /// Check whether `now` is after or equal to `when`, w.r.t. a `reference` time point.
+    fn expired(reference: Self, now: Self, when: Self) -> bool;
```

Comment:
Isn't the implementation for this always the same? Should this be a helper instead of a method? also, it might be more clear what each argument means if it is named slightly differently:

```rust
fn gt_or_eq<T: Ticks>(reference: T, lhs: T, rhs: T) -> bool
```

Reply:
I agree that it's not clear what it would mean for a number of bits (or counter width) to be expired.